### PR TITLE
feat: Introduce compose logs subcommand

### PR DESCRIPTION
**Pull request by LucaSeri**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes
Introduce the ``kraft compose logs`` subcommand that allows users to print existing logs or follow them.

E.g:

```bash
kraft compose up -d

kraft compose logs --follow

ping-pong-pong  | Received: ping
ping-pong-ping  | Received: pong
ping-pong-pong  | Received: ping
```

### CODE REVIEW

File: internal/cli/kraft/compose/logs/logs.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+	if err := project.Validate(ctx); err != nil {
+		return err
+	}
+
+	return opts.PrintLogs(ctx, project)
+}
+
+func (opts *LogsOptions) PrintLogs(ctx context.Context, project *compose.Project) error {
+	var wg sync.WaitGroup
```
Comment (craciunoiuc) [inline internal/cli/kraft/compose/logs/logs.go]: shouldn't this be private? Is it used anywhere else?

`printLogs`

File: internal/cli/kraft/compose/logs/logs.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+	longestName := 0
+	for _, service := range project.Services {
+		if len(service.Name) > longestName {
+			longestName = len(service.Name)
+		}
+	}
+	for i := range project.Services {
```
Comment (craciunoiuc) [inline internal/cli/kraft/compose/logs/logs.go]: wouldn't it look better to just do:

`for _, service := range project.Services {`

and then

`(service)`

?

Comment (craciunoiuc) [reply internal/cli/kraft/compose/logs/logs.go]: Actually even better, only the name seems to be used, so

`(service.Name)` and then use a `name` variable inside everywhere

File: internal/cli/kraft/compose/logs/logs.go (hunk 1 above)
Comment (craciunoiuc) [reply internal/cli/kraft/compose/logs/logs.go]: Nvm, saw it used  in some other place also below.

Move it please to a `internal/cli/kraft/compose/utils/utils.go` file and make public there

**Review by craciunoiuc — CHANGES_REQUESTED** [review]
left you some comments inline, overall looks fine

**Comment by LucaSeri** [comment]
Thanks @craciunoiuc for the comments. Part of this PR was actually migrated to https://github.com/unikraft/kraftkit/pull/1402, and ``kraft compose logs`` is now almost trivial.

**Review by nderjung — APPROVED** [review]
Thanks!

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Pull request merged**